configuration/introduction.md

```diff
@@ -23,3 +23,176 @@ The following properties can be configured:
 | `customCss`        | The path of the `custom.css` stylesheet. The default is `css/custom.css`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
+
+### Environment variables
+
+As the `config.js` is javascript code there is no direct way to provide environment variables. This feature was requested by some users, e.g. with environment variables it is easier to share the configuration with others (otherwise you have to delete all private data as e.g. api keys). There was a longer [discussion on github](https://github.com/MichMich/MagicMirror/issues/1756) where you can read the full story.
+
+#### `config.js` and `config.js.template`
+
+You can provide a `config.js.template` instead of a `config.js` file. The difference is, that you can use environment variables in the `config.js.template`. When starting MagicMirror² a `config.js` is created from `config.js.template` and the variables are resolved.
+
+Variables must be inserted as `${MY_VARIABLE}`, examples:
+
+`config.js.template`:
+```js
+let config = {
+	address: "${MY_ADDRESS}",
+	port: ${MY_PORT},
+	useHttps: ${MY_HTTPS},
+};
+/*************** DO NOT EDIT THE LINE BELOW ***************/
+if (typeof module !== "undefined") {module.exports = config;}
+```
+
+would become
+
+`config.js`:
+```js
+let config = {
+	address: "localhost",
+	port: 8080,
+	useHttps: false,
+};
+/*************** DO NOT EDIT THE LINE BELOW ***************/
+if (typeof module !== "undefined") {module.exports = config;}
+```
+
+#### Providing environment variables
+
+There are 2 ways for declaring them, you can mix both. If a variable is declared in both ways the one declared as linux environment variable is used.
+
+##### Using a `config.env` file
+
+This file must be in the same folder as the `config.js.template` and contains the variables, example from above:
+
+File content of `config.env`:
+
+```bash
+MY_ADDRESS=localhost
+MYPORT=8080
+MY_HTTPS=false
+```
+
+##### Using linux environment variables
+
+You have to declare them before you start MagicMirror², e.g. by executing:
+
+```bash
+export MY_ADDRESS=localhost
+export MYPORT=8080
+export MY_HTTPS=false
+```
+
+#### Real world example of `config.js.template`
+
+```js
+var config = {
```

Comment:
var -> let

Reply:
thanks ... may I should update my config ;)

fixed.